interface.go

```diff
@@ -16,33 +17,49 @@ type TableInfo struct {
 }
 
 type ColumnDef struct {
-	Path string    `json:"path"`
+	Name string    `json:"name"`
+	Key  string    `json:"key"`
 	Type FieldType `json:"type"`
 }
 
 type TableDriver interface {
 	New(name string, columns []ColumnDef) (TableStore, error)
 	Get(name string) (TableStore, error)
+	GetAllColNames() chan string
+	GetLabels(edges bool) chan string
 	List() []string
+	Delete(name string) error
 	Close()
 }
 
-type Entry struct {
-	Key   []byte
-	Value map[string]any
+type Row struct {
+	Id    []byte
+	Label string
+	Data  map[string]any
 }
 
-type TableStore interface {
-	GetColumns() []ColumnDef
-	Add(key []byte, row map[string]any) error
-	Get(key []byte, fields ...string) (map[string]any, error)
-	Delete(key []byte) error
+type Index struct {
+	Key      []byte
+	Position uint64
+}
 
-	Scan(filter []FieldFilter, fields ...string) chan map[string]any
+type BulkResponse struct {
+	Key  string
```

Comment:
Is the Key a `string` or a `[]byte`

Reply:
Key could be string or byte. Looks like in most of the funcs it has to convert byte to string to fit this definition. Might be better to leave that up to downstream functions to figure out.